### PR TITLE
Tests for parse_result

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Json parser returns error if parse
fail. Add test to check parsing success
with valid responce and fails with
invalid json.

Relates-To: OLPEDGE-2147

Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

**Review by kzvieriev — APPROVED** [review]

File: olp-cpp-sdk-dataservice-read/tests/JsonResultParserTest.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    auto responce = olp::parser::parse_result<ExtendedResponse>(str, 100);
+    ASSERT_FALSE(responce.IsSuccessful());
+    ASSERT_EQ(responce.GetError().GetErrorCode(),
+              olp::client::ErrorCode::Unknown);
+    ASSERT_EQ(responce.GetError().GetMessage(), "Fail parsing response.");
+  }
+}
+
+TEST(JsonResultParserTest, WithArgs) {
```
Comment (meravingen123) [inline olp-cpp-sdk-dataservice-read/tests/JsonResultParserTest.cpp:115]: This is basically the same as ExtendedResponse

File: olp-cpp-sdk-dataservice-read/tests/JsonResultParserTest.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+#include "JsonResultParser.h"
+#include "generated/serializer/PartitionsSerializer.h"
+#include "generated/serializer/JsonSerializer.h"
+#include "ReadDefaultResponses.h"
+#include "ExtendedApiResponse.h"
+// clang-format on
+
+namespace dr = olp::dataservice::read;
+
+TEST(JsonResultParserTest, Partitions) {
```
Comment (meravingen123) [inline olp-cpp-sdk-dataservice-read/tests/JsonResultParserTest.cpp:36]: Maybe rename this to just 'ParseResult' ?

File: olp-cpp-sdk-dataservice-read/tests/JsonResultParserTest.cpp (hunk 1 above)
Comment (LiubovDidkivska) [reply olp-cpp-sdk-dataservice-read/tests/JsonResultParserTest.cpp:115]: It not the same, ExtendedResponse has additional arg not as Result, here we have Result that combined from parsed type and other types.

File: olp-cpp-sdk-dataservice-read/tests/CMakeLists.txt [hunk 3]
```diff
@@ -24,6 +24,7 @@ set(OLP_SDK_DATASERVICE_READ_TEST_SOURCES
     DataRepositoryTest.cpp
     MetadataApiTest.cpp
     ParserTest.cpp
+    JsonResultParserTest.cpp
```
Comment (meravingen123) [inline olp-cpp-sdk-dataservice-read/tests/CMakeLists.txt:27]: We use alphabetical order here

**Review by meravingen123 — APPROVED** [review]

**Review by kzvieriev — APPROVED** [review]

**Pull request merged**